### ISSUE TITLE
 sector stuck in PreCommit2 / handler: websocket connection closed / ERROR: normal shutdown of state machine

### ISSUE THREAD

**Issue by RobQuistNL** [filecoin-project/lotus#3451]
**Describe the bug**
A sector is stuck in PreCommit2

**To Reproduce**
Pledge sector, wait

**Expected behavior**
Sector getting sealed

**Screenshots**
```
lotus-miner sectors status --log 59
SectorID:	59
Status:		PreCommit2
CIDcommD:	baga6ea4seaqao7s73y24kcutaosvacpdjgfe5pw76ooefnyqw4ynr3d2y6x2mpq
CIDcommR:	bagboea4b5abcalfspzsfbmvwzxqc4vuilsqqq5nas2jd7okhpgozsq3loheywlcw
Ticket:		8ec298ec7e7a65d3e0fb9336d48de8dc47fa1fa139b1e08f8e37bb845b992d34
TicketH:	17095
Seed:		
SeedH:		0
Precommit:	bafy2bzacedrurjsciiagoft4kvb35jiydxf45cdykhymcxk5a2p5uufdyl4fe
Commit:		<nil>
Proof:		
Deals:		[0]
Retries:	0
--------
Event Log:
0.	2020-08-31 03:57:54 +0000 UTC:	[event;sealing.SectorStartCC]	{"User":{"ID":59,"SectorType":3,"Pieces":[{"Piece":{"Size":34359738368,"PieceCID":{"/":"baga6ea4seaqao7s73y24kcutaosvacpdjgfe5pw76ooefnyqw4ynr3d2y6x2mpq"}},"DealInfo":null}]}}
1.	2020-08-31 03:57:54 +0000 UTC:	[event;sealing.SectorPacked]	{"User":{"FillerPieces":null}}
2.	2020-08-31 09:37:01 +0000 UTC:	[event;sealing.SectorPreCommit1]	{"User":{"PreCommit1Out":"eyJ(...)FuZLTQ=","TicketEpoch":17095}}
3.	2020-08-31 11:36:14 +0000 UTC:	[event;sealing.SectorPreCommit2]	{"User":{"Sealed":{"/":"bagboea4b5abcalfspzsfbmvwzxqc4vuilsqqq5nas2jd7okhpgozsq3loheywlcw"},"Unsealed":{"/":"baga6ea4seaqao7s73y24kcutaosvacpdjgfe5pw76ooefnyqw4ynr3d2y6x2mpq"}}}
4.	2020-08-31 11:43:16 +0000 UTC:	[event;sealing.SectorPreCommitted]	{"User":{"Message":{"/":"bafy2bzacedrurjsciiagoft4kvb35jiydxf45cdykhymcxk5a2p5uufdyl4fe"},"PreCommitDeposit":"641291675567200817","PreCommitInfo":{"SealProof":3,"SectorNumber":59,"SealedCID":{"/":"bagboea4b5abcalfspzsfbmvwzxqc4vuilsqqq5nas2jd7okhpgozsq3loheywlcw"},"SealRandEpoch":17095,"DealIDs":[],"Expiration":1570489,"ReplaceCapacity":false,"ReplaceSectorDeadline":0,"ReplaceSectorPartition":0,"ReplaceSectorNumber":0}}}
5.	2020-08-31 15:23:36 +0000 UTC:	[event;sealing.SectorChainPreCommitFailed]	{"User":{}}
	handler: websocket connection closed
6.	2020-08-31 19:14:29 +0000 UTC:	[event;sealing.SectorForceState]	{"User":{"State":"PreCommit2"}}
7.	2020-08-31 19:52:45 +0000 UTC:	[event;sealing.SectorSealPreCommit2Failed]	{"User":{}}
	seal pre commit(2) failed: failed to acquire sector {2388 59} from remote (tried [{f8659919-bfe2-4ec2-b329-b2b2b2b2b2b2 [http://192.168.2.4:3456/remote/cache/s-t02388-59] 10 true false true}]): 1 error occurred:
	* fetch error http://192.168.2.4:3456/remote/cache/s-t02388-59 (storage f8659919-bfe2-4ec2-b329-b2b2b2b2b2b2) -> /mnt/nvme_2tb_slow/lotus-calibration/cache/fetching/s-t02388-59: http: unexpected EOF reading trailer
8.	2020-08-31 19:53:45 +0000 UTC:	[event;sealing.SectorRetrySealPreCommit2]	{"User":{}}
9.	2020-08-31 19:53:45 +0000 UTC:	[event;sealing.SectorSealPreCommit2Failed]	{"User":{}}
	seal pre commit(2) failed: failed to acquire sector {2388 59} from remote (tried [{f8659919-bfe2-4ec2-b329-b2b2b2b2b2b2 [http://192.168.2.4:3456/remote/cache/s-t02388-59] 10 true false true}]): 1 error occurred:
	* fetch error http://192.168.2.4:3456/remote/cache/s-t02388-59 (storage f8659919-bfe2-4ec2-b329-b2b2b2b2b2b2) -> /mnt/nvme_2tb_slow/lotus-calibration/cache/fetching/s-t02388-59: do request: Get "http://192.168.2.4:3456/remote/cache/s-t02388-59": dial tcp 192.168.2.4:3456: connect: connection refused


10.	2020-08-31 19:54:45 +0000 UTC:	[event;sealing.SectorForceState]	{"User":{"State":"PreCommit2"}}
11.	2020-08-31 19:54:45 +0000 UTC:	[event;sealing.SectorRetrySealPreCommit1]	{"User":{}}
```

**Version:** 0.5.6

**Additional context**
Trying to reset the state:
```
$ lotus-miner sectors update-state --really-do-it 59 WaitSeed
ERROR: normal shutdown of state machine
$ lotus-miner sectors update-state --really-do-it 59 PreCommit2
ERROR: normal shutdown of state machine
```

**Comment by Reiers** [comment]
**Hi @RobQuistNL** 

Thanks for the report. This is fixed in the latest release.
Please upgrade Lotus.

Link to latest tags [here.](https://github.com/filecoin-project/lotus/tags)

```
v1.14.4 -  SPs should be able to force
update sectors status without getting blocked by normal shutdown of state machine.
```

Closing Lotus issue ticket.

If you are still experiencing problems, open a new ticket or ask for help in our [Discussions forum.](https://github.com/filecoin-project/lotus/discussions)

**Comment by RobQuistNL** [comment]
This is still happening in 1.17.1, I'm not getting errors but the sectors remain stuck.

**Comment by RobQuistNL** [comment]
Hmm, it looks like it took 6 remove & terminate commands, 2 miner restarts and some worker restarts, but sectors finally started disappearing as I was writing this comment :)
